exercises/ayomide_bamidele/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,138 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <map>
+#include <algorithm>
+#include <cctype>
+#include <vector>
+#include <iomanip>
+#include <utility>
+
+
+using std::string; using std::max; using std::find_if_not;
+using std::ifstream; using std::fstream;
+using std::getline; using std::map; using std::pair; using std::transform;
+ using std::endl; using std::vector;
+using std::cout; using std::find; using std::ofstream;
+using std::find_if; using std::vector; using std::setw;
+
+bool pass(string str, string sample, size_t limit);
+
+int main(){
+	// vector of pairs to sort
+	vector < pair < string, int > > v;
+
+    // max word length
+    size_t size = 0;
+
+    //write each word into line
+    string line;
+
+    // create input file object
+    ifstream infile("Holmes.txt");
+
+	// create output file
+	ofstream outfile("Holmes Counter.txt");
+
+    // map to store word occurences
+    map<string, int> occurences;
+    
+    // open file for reading
+    if (infile.is_open()){
+
+        // read line
+        while(getline(infile, line)){
+            
+            // iterator to loop through each line
+            auto i = line.begin();
+
+            while(i != line.end()){
+				
+                // ignore leading blanks
+				i = find_if_not(i, line.end(), ::isspace);
+
+				// ignore leading characters
+				i = find_if(i, line.end(), ::isalpha);
+
+				// if line is all whitespaces or has reached the end
+				if (i == line.end()) {
+					break;
+				}
+
+                // find end of next word
+                auto j = find_if(i, line.end(), ::isspace);
+
+                // copy the characters in [i, j) and increment the counter
+                // or add to occurences after checking conditions
+				if (i != line.end()) {
+
+					// word
+					string word = string(i, j);
```

Comment:
Do you mean instead of using string?

Reply:
Yes. On the right you are constructing a string, so the compiler should be able to deduce the type of `word` from this.